pkg/operator/controllers/storageaccounts/storageaccounts.go

```diff
@@ -37,6 +37,13 @@ func (r *reconcileManager) reconcileAccounts(ctx context.Context) error {
 		rc.Spec.Storage.Azure.AccountName,
 	}
 
+	// If managed=true, set default action to Deny
+	// If managed=false, set default action to Allow
+	defaultAction := mgmtstorage.DefaultActionAllow
+	if managed {
```

Comment:
If we don't manage the storage account (which means the default action is Allow), we still set the `VirtualNetworkRules` as it simplifies the logic and won't matter as it's allowed anyway.